### ISSUE TITLE
Discussion: Lit-Element for Justgage

### ISSUE THREAD

**Issue by robertsLando** [toorshia/justgage#333]
Hi @toorshia and @deezone, I have found [this](https://github.com/thomasloven/round-slider/) interesting project on github that uses [Lit-Element](https://lit-element.polymer-project.org/) that is 

> A simple base class for creating fast, lightweight web components

It is really interesting because it allows to write really clean code for web components and maybe could allow us to also remove Raphael as dependence. 

Any thoughts about it? The example I have linked shows a round slider that is not so far from this.

**Comment by toorshia** [comment]
Hey @robertsLando. Thanks for researching into this. I like the idea of removing Raphael as  a dependence. I am wondering if Lit-Element can provide us with the drawing features Raphael does? Can you research into pros/cons? I totally support this initiative, unless it limits us in achieving stuff from roadmap for 2.0, and future developments.

**Comment by deezone** [comment]
@robertsLando did you see the demo that @robertsLando is pointing to: https://github.com/thomasloven/round-slider

In effect, the demo shows how Lit-Element supports SVG (https://www.w3.org/Graphics/SVG/). You can see the setup here:
```
import {
  LitElement,
  html,
  css,
  svg,
} from "lit-element";
```
https://github.com/thomasloven/round-slider/blob/master/src/main.js#L1-L6

And where the _magic_ happens:
https://github.com/thomasloven/round-slider/blob/master/src/main.js#L295-L336

Looks great to me!

**Comment by robertsLando** [comment]
Yes it looks amazing, we should also check if it is possibile to get events like Raphael does. 

We could start a dev branch with the porting to LitElement? 

Daniel

**Comment by deezone** [comment]
Ooops, I flagged @robertsLando when I meant @toorshia  :)

What are your thoughts on my comments @toorshia ?
https://github.com/toorshia/justgage/issues/333#issuecomment-560037230

**Comment by toorshia** [comment]
@deezone just saw it, I see in example he wrote his custom "renderArc" method, right? Does this mean we would have to re-create all the drawing methods rom Raphael we are currently using?

**Comment by robertsLando** [comment]
Yes but should not be that hard in my opinion, the only thing I actually don’t know is how to support Raphael events we currently bind too without using Raphael 

Daniel

**Comment by toorshia** [comment]
Allrighty then, let's start a dev branch for LitElement! 👍 @robertsLando can you take a crack at this?

**Comment by robertsLando** [comment]
I think that we could start another repo because it would be a complete rewrite, almost everything will change in the code. What about a Justgage2?

Daniel

**Comment by toorshia** [comment]
Whatever you say, chief. You're the engineering brains around here :)

**Comment by robertsLando** [comment]
Unfortunally now I would not have much time to spend on it because I have many others repo to keep updated, I'm focusing on fixing bugs/compatibility issues but no time for new features/projects right now. Let's keep this open for the future, I would be glad to implement this :)

**Comment by toorshia** [comment]
Sure thing 👍:)
